### PR TITLE
Soarian datetime changes

### PR DESCRIPTION
**Pull request by amarvakul**
@MaxPhilips @daliboz @slvsridhar 
We made some edits to the notes section pertaining to date parameter behaviors to some resources, please review.

Thanks,
Doug & Amaresh

### CODE REVIEW

File: content/soarian/dstu2/care-provision/careplan.md [hunk 1]
```diff
@@ -55,11 +55,11 @@ _Implementation Notes_
 Notes:
 
 * See details regarding special [date][date-parameter] requirements.
-* If the date parameter is null in the query, the API will return qualifying records with dates less than or equal to the date of the query as limited by the lesser of 1000 records or 30 days inclusive of the query date.
-* If a single date parameter is used in the query, the response will include qualifying records dated that day as limited by 1000 records.
-* If a valid date range is used in the query, the API will return qualifying records within the dates specified as limited by the lesser of 1000 records or 30 days starting at the upper limit of the range.
-* If a date range greater than 30 days is used in the query, the response will include qualifying records within the dates specified as limited by the lesser of 1000 records or 30 days starting at the upper limit of the range. In this case, the informational response in the narrative extension will indicate that the date range is greater than 30 days and should be refined.
+* If the date parameter is null in the query, the API will return qualifying records.
```
Comment (MaxPhilips) [inline content/soarian/dstu2/care-provision/careplan.md]: To be consistent with other changes: "...the API will return *all* qualifying records."

File: content/soarian/dstu2/diagnostic/diagnostic-report.md [hunk 2]
```diff
@@ -49,11 +49,11 @@ _Implementation Notes_
...
-* If the query date parameter is null in the query, the API will return qualifying records with dates less than or equal to the date of query as limited by the lesser of 1000 records or 30 days inclusive of the query date.
-* If a single date parameter is included, the response will include qualifying records dated that day as limited by 1000 records.
-* If a valid date range is used in the query, the API will return qualifying records within the dates specified as limited by the lesser of 1000 records or 30 days starting at the upper limit date specified.
-* If a date range is used greater than 30 days, the response will include qualifying records within the dates specified as limited by the lesser of 1000 records or 30 days starting at the upper limit date specified and include in the informational response in the narrative extension that the date range entered is greater than 30 days and please refine your response.
-* If an invalid date range is specified in the query, the API will error.
+* If the query date parameter is null in the query, the API will return all qualifying records.
```
Comment (MaxPhilips) [inline content/soarian/dstu2/diagnostic/diagnostic-report.md]: To be consistent with other changes: "If the ~~query~~ date parameter..."

File: content/soarian/dstu2/diagnostic/diagnostic-report.md [hunk 3]
```diff
@@ -49,11 +49,11 @@ _Implementation Notes_
...
+* If a single date parameter is included, the response will include all qualifying records dated that day.
+* If a valid date range is used in the query, the API will return all qualifying records within the dates specified.
+* If an invalid date range is used in the query, the API will error with code [500] [common-errors]. Please refer to [Special information regarding date parameters] [date-parameter].
```
Comment (daliboz) [inline content/soarian/dstu2/diagnostic/diagnostic-report.md]: These links don't work - you need to add them at the bottom of the file just like CarePlan

**Comment by daliboz** [comment]
+1 after changes above are made.

File: content/soarian/dstu2/care-provision/careplan.md (hunk 1 above)
Comment (MaxPhilips) [reply content/soarian/dstu2/care-provision/careplan.md]: https://github.com/cerner/fhir.cerner.com/pull/187/commits/8aee949bc8c2cf0964f281a84c81ffa5b08fdfa2

File: content/soarian/dstu2/diagnostic/diagnostic-report.md (hunk 2 above)
Comment (MaxPhilips) [reply content/soarian/dstu2/diagnostic/diagnostic-report.md]: https://github.com/cerner/fhir.cerner.com/pull/187/commits/6f242cdf9dfd1ab8fd6678925e88e2bf12c5a22c

**Comment by daliboz** [comment]
@amarvakul The latest changes crash nanoc for me/I can't generate the site. Looks like it's an issue in doc ref:

NameError: uninitialized constant Cerner::Resources::SOARIAN_DOCUMENT_REFERENCE_DOCREF_BUNDLE_ATTACHMENT
Did you mean?  Cerner::Resources::SOARIAN_DOCUMENT_REFERENCE_DOCREF_BUNDLE

Also, all the examples that were updated removed the error extensions, and populated the coding ("what" the resource is) with a null/unknown value where it had loinc/was coded properly before. Is this what the API will actually do? Or just an issue with testing data?

**Comment by MaxPhilips** [comment]
Closing in favor of https://github.com/cerner/fhir.cerner.com/pull/187

**Pull request closed without merging**